grafe-cli/src/generate/generate.ts

```diff
@@ -3,83 +3,176 @@ import yargs, { Options } from 'yargs';
 import fs from 'fs';
 import mkdirp from 'mkdirp';
 import pkgDir from 'pkg-dir';
-
-export function generateCommand(yargs: yargs.Argv<{}>) {
-    return yargs;
+import * as path from 'path';
+
+export function generateCommand(yargs: yargs.Argv<{}>): yargs.Argv<{}> {
+    return yargs.positional('type', {
+        description: 'Type of component you want to generate',
+        choices: ['route', 'middleware']
+    }).option('routePath', {
+        alias: 'r',
+        type: 'string',
+        description: 'Path of the new route'
+    }).option('method', {
+        alias: 'm',
+        type: 'string',
+        description: 'HTTP-Method of the route'
+    }).option('middlewares', {
+        alias: 'w',
+        type: 'array',
+        description: 'Middlewares for this route'
+    }).option('name', {
+        alias: 'n',
+        type: 'string',
+        description: 'Name of the middleware'
+    }).option('short', {
+        alias: 's',
+        type: 'string',
+        description: 'Short name of the middleware'
+    }).option('description', {
+        alias: 'd',
+        type: 'string',
+        description: 'Description of the middleware'
+    });
 }
 
 /**
  * Starts the prompt for generating a new component
  */
-export function generateCLI() {
-    inquirer.prompt([
-        {
-            type: 'list',
-            name: 'type',
-            message: 'What do you want to generate',
-            choices: ['Route', 'Middleware'],
-        }
-    ]).then(async (answers) => {
-        if (answers.type === 'Route') {
-
-            const rootDir = await pkgDir(process.cwd());
-
-            let raw;
-            try {
-                raw = fs.readFileSync(rootDir + "/grafe.json");
-            } catch (err) {
-                console.log("The grafe command must be used within a grafe project.");
-                return;
+export async function generateCLI(argv: any): Promise<void> {
+
+    let type: string = argv.type;
+    
+    let answers;
+    if (type == undefined || type.length == 0) {
+        answers = await inquirer.prompt([
+            {
+                type: 'list',
+                name: 'type',
+                message: 'What do you want to generate',
+                choices: ['Route', 'Middleware'],
             }
+        ]);
+    } else {
+        if (type.toLocaleLowerCase() == 'middleware' || type.toLocaleLowerCase() == 'route') {
+            answers = { type: type };
```

Comment:
Ich hätt hier gleich 
```suggestion
            answers = { type: type.toLocaleLowerCase() };
```
gemacht dann hättest es unten ned immer amchen müssen